pkg/virtctl/imageupload/imageupload.go

```diff
@@ -217,7 +217,14 @@ func (c *command) run(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
-	defer util.CloseIOAndCheckErr(file, nil)
+
+	// on successful upload, file is closed by http transferWriter via BodyCloser, so let's prevent double-close
+	dataUploaded := false
+	defer func() {
+		if !dataUploaded {
```

Comment:
It may make sense to keep the close in this function and pass in `uploadData` a `ioutil.NopCloser(file)`  to the http request, to ensure that we have full control about the close process.

Reply:
Right, much smarter. Done, thanks!

Reply:
@rmohr The only issue I see is that now the actual close is done at the end of the calling function. Need to be carful not to call the uploadData twice before closing the reader externally. As this is a local function which is used from a single place/thread - we are probably OK.